### ISSUE TITLE
Dragging over the edge of the paper selects stuff (Chrome only)

### ISSUE THREAD

**Issue by nomeata** [clientIO/joint#181]
My users that are using Chrome [report](https://github.com/nomeata/incredible/issues/17) this undesirable behaviour: When starting a mouse drag on the paper, but then crossing the edge of the paper, the browser starts to select stuff there, which looks really weird. Can be reproduced on, e.g., http://www.jointjs.com/demos/logic as well.

On Firefox, this works as expected.

Not sure if there is an easy fix. If all browsers would support [`setCapture`](https://developer.mozilla.org/docs/Web/API/Element/setCapture), that might be a solution, but I believe Chrome does not support this.

**Comment by chill117** [comment]
I did these steps:
* Opened http://www.jointjs.com/demos/logic
* Click + hold the left mouse button at the center of the paper
* Drag the cursor outside of the edge of the paper

This results in a selection occurring on HTML elements outside the paper element. This exact behavior happens in Chrome, Firefox, and Internet Explorer. What is the expected behavior?

**Comment by nomeata** [comment]
Note that with Firefox, if you click and hold an element, it will _not_ select any of the HTML elements outside the paper. I would consider that the expected behaviour.

**Comment by chill117** [comment]
> Note that with Firefox, if you click and hold an element, it will not select any of the HTML elements outside the paper. I would consider that the expected behaviour.

I was not able to reproduce the case where no HTML elements are selected outside paper (the one you describe as the desired behavior). I tried these browsers:
* Firefox 41 on Ubuntu
* Chrome 46 on Ubuntu
* IE 10 on Windows 7
* Firefox 38 on Windows 7
* Chrome 46 on Windows 7
* Opera 33 on Windows 7

**Comment by chill117** [comment]
I've looked at this closely. I don't know if this is something we can fix within joint itself. Since it's not any joint HTML/SVG elements being selected, but your UI elements outside the paper. Have you tried using [user-select](https://css-tricks.com/almanac/properties/u/user-select/) for the UI elements being selected?

**Comment by nomeata** [comment]
No, I didn’t do any work on this, sorry.

**Comment by DavidDurman** [comment]
This can and should be fixed inside JointJS. We should be fine by calling `evt.preventDefault()` in `joint.dia.element.js` when the dragging starts (in `pointerdown()` or `pointermove()` methods).

Before that, you can set `user-select` to `none` in your CSS e.g. for the `body` or paper container as @chill117 pointed out.

**Comment by chill117** [comment]
Fixed this upstream. Should be included in the next joint release.

**Comment by elyas-bhy** [comment]
@chill117 can we get a hotfix for this issue in 0.9.5? When is the expected date for the next release?

**Comment by chill117** [comment]
@elyas-bhy Sure thing

Change the following:
```js
    pointerdown: function(evt) {

        evt = joint.util.normalizeEvent(evt);

        var view = this.findView(evt.target);
        if (this.guard(evt, view)) return;

        var localPoint = this.snapToGrid({ x: evt.clientX, y: evt.clientY });

        if (view) {

            this.sourceView = view;

            view.pointerdown(evt, localPoint.x, localPoint.y);

        } else {

            this.trigger('blank:pointerdown', evt, localPoint.x, localPoint.y);
        }
    },
```

To:
```js
    pointerdown: function(evt) {

        evt = joint.util.normalizeEvent(evt);

        var view = this.findView(evt.target);
        if (this.guard(evt, view)) return;

        evt.preventDefault();

        var localPoint = this.snapToGrid({ x: evt.clientX, y: evt.clientY });

        if (view) {

            this.sourceView = view;

            view.pointerdown(evt, localPoint.x, localPoint.y);

        } else {

            this.trigger('blank:pointerdown', evt, localPoint.x, localPoint.y);
        }
    },
```

That should do the trick.

**Comment by kumilingus** [comment]
Fixed in v0.9.6.
